crypto/groups/qr_special_rsa.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 XLAB d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package groups
+
+// TODO: move qr.go into groups package, make QR parent of QRSpecialRSA
+// TODO: check group_generators.go and move the content into groups
+
+import (
+	"github.com/xlab-si/emmy/crypto/common"
+	"math/big"
+)
+
+// QRSpecialRSA presents group of quadratic residues modulo N where N is a product
+// of two safe primes.
+type QRSpecialRSA struct {
+	N  *big.Int // N = P * Q, P = 2*p + 1, Q = 2*q + 1
+	P *big.Int
+	Q *big.Int
+	SmallP *big.Int
+	SmallQ *big.Int
+	Order *big.Int // order of group QR_N (it is SmallP * SmallQ)
+}
+
+func NewQRSpecialRSA(safePrimeBitLength int) (*QRSpecialRSA, error) {
+	P, Q, p, q, err := common.GetSpecialRSAPrimes(safePrimeBitLength)
+	if err != nil {
+		return nil, err
+	}
+	N := new(big.Int).Mul(P, Q)
+	order := new(big.Int).Mul(p, q)
```

Comment:
I would move right-hand side of expressions on L44 and L45 to L47 and L62, respectively.